Sources/TuistGraph/Models/Arguments.swift

```diff
@@ -19,3 +19,32 @@ public struct Arguments: Equatable, Codable {
         self.launchArguments = launchArguments
     }
 }
+
+extension Arguments: Equatable {
+    /// Implement `Equatable` manually so order of arguments doesn't matter.
+    public static func == (lhs: Arguments, rhs: Arguments) -> Bool {
+        lhs.environment == rhs.environment
+            && lhs.launchArguments.sorted { $0.name < $1.name }
+            == rhs.launchArguments.sorted { $0.name == $1.name }
+    }
+}
+
+extension Arguments {
+    /// Creates a new `Arguments` that merges the contents of the current and given `Arguments`.
+    ///
+    /// If there are duplicate keys, the value of the current one will be preserved.
+    ///
+    /// - Parameter arguments: The `Arguments` to merge.
+    /// - Returns: A new `Arguments` with the merged contents.
+    public func merging(with arguments: Arguments) -> Arguments {
```

Comment:
Let's move this where it's actually needed (i.e. in the Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift file)